### DISCUSSION TITLE
Edit Current of an Existing Winding Group

### DISCUSSION THREAD

**Discussion by nikmarlo** [ansys/pyaedt#5170] [Q&A] [answered]
Hey,

this problem seems basic to me, but I am having trouble editing the properties of existing winding groups.

My winding groups are defined using:

```
winding_group_p1 = "Primary_1"
winding_group_p2 = "Primary_2"
winding_group_s = "Secondary"

maxwell_2d.assign_winding(winding_type="Current", is_solid=True, current=1, name=winding_group_p1)
maxwell_2d.assign_winding(winding_type="Current", is_solid=True, current=1, name=winding_group_p2)
maxwell_2d.assign_winding(winding_type="Current", is_solid=True, current=1, name=winding_group_s)
```
To edit the current afterwards, my first attempt was using `assign_current()`, but it is obviously not applicable in my case.

So I tried recording the script, which gave me this result:
```
import ScriptEnv
ScriptEnv.Initialize("Ansoft.ElectronicsDesktop")
oDesktop.RestoreWindow()
oProject = oDesktop.SetActiveProject("Project_NJB")
oDesign = oProject.SetActiveDesign("Maxwell 2D_9ZK")
oDesign.ChangeProperty(
	[
		"NAME:AllTabs",
		[
			"NAME:Maxwell2D",
			[
				"NAME:PropServers", 
				"BoundarySetup:Primary_1"
			],
			[
				"NAME:ChangedProps",
				[
					"NAME:Current",
					"Value:="		, "20A"
				]
			]
		]
	])
oDesign.ChangeProperty(
	[
		"NAME:AllTabs",
		[
			"NAME:Maxwell2D",
			[
				"NAME:PropServers", 
				"BoundarySetup:Primary_1"
			],
			[
				"NAME:ChangedProps",
				[
					"NAME:Phase",
					"Value:="		, "1.5deg"
				]
			]
		]
	])
oDesign.ChangeProperty(
	[
		"NAME:AllTabs",
		[
			"NAME:Maxwell2D",
			[
				"NAME:PropServers", 
				"BoundarySetup:Primary_1"
			],
			[
				"NAME:ChangedProps",
				[
					"NAME:Phase",
					"Value:="		, "1.5rad"
				]
			]
		]
	])
```

However, I am afraid I am not familiar enough with the PYAEDT syntax to translate this code. Sorry if I just didn't find the corresponding documentation.

Thanks a lot!

**Comment by Samuelopez-ansys** [comment, marked as answer]
@nikmarlo One of the main advantages of using PyAEDT instead of the AEDT API, is the object oriented approach.

You can save the returned object in a variable like:

wind1 = maxwell_2d.assign_winding(winding_type="Current", is_solid=True, current=1, name=winding_group_p1)

If you do dir(wind1) you will see all the properties available, they are getter and setter. For instance:

wind1.object_properties.props["Current"] = "1A"

**Reply by nikmarlo** [reply]
Awesome, thank you Samuel!
